lang_php/parsing/parser_php.mly

```diff
@@ -74,7 +74,10 @@ let mk_param s =
 let mk_var (s, tok) =
   match s with
   | "this" -> This tok
-  | _ -> IdVar (DName(s, tok))
+  | _ ->
+    (* assume variable is a metavariable if it is all uppercase *)
+    if String.uppercase_ascii s = s then MetaVar (DName(s, tok))
```

Comment:
You can use AST_generic_helpers.is_metavar_name ("$" ^ s)

Reply:
And you need to make sure we do that only when Flag_parsing.sgrep_mode is true.

As I said, I'd rather not add a Metavar but at more places where we match Id, we should add match
for Var with the is_metavar_name && sgrep_mode attached to it.